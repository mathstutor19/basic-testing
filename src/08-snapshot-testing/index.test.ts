import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  test('should generate linked list from set of values object snapshot', () => {
    const elements = [1, 2, 3];
    const result = generateLinkedList(elements);

    expect(result).toMatchSnapshot();
  });

  test('should generate linked list from different set of values object snapshot', () => {
    const elements = ['a', 'b', 'c'];
    const result = generateLinkedList(elements);

    expect(result).toMatchSnapshot();
  });

  test('should generate correct linked list from numeric values', () => {
    expect(generateLinkedList([1])).toStrictEqual({
      value: 1,
      next: { value: null, next: null },
    });

    expect(generateLinkedList([1, 2])).toStrictEqual({
      value: 1,
      next: { value: 2, next: { value: null, next: null } },
    });

    expect(generateLinkedList([1, 2, 3])).toStrictEqual({
      value: 1,
      next: { value: 2, next: { value: 3, next: { value: null, next: null } } },
    });
  });

  test('should generate correct linked list from numeric values snapshot', () => {
    expect(generateLinkedList([1])).toMatchSnapshot();
    expect(generateLinkedList([1, 2])).toMatchSnapshot();
    expect(generateLinkedList([1, 2, 3])).toMatchSnapshot();
  });
});
