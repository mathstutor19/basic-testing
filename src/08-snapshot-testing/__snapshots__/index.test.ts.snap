// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate correct linked list from numeric values snapshot 1`] = `
{
  "next": {
    "next": null,
    "value": null,
  },
  "value": 1,
}
`;

exports[`generateLinkedList should generate correct linked list from numeric values snapshot 2`] = `
{
  "next": {
    "next": {
      "next": null,
      "value": null,
    },
    "value": 2,
  },
  "value": 1,
}
`;

exports[`generateLinkedList should generate correct linked list from numeric values snapshot 3`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": 3,
    },
    "value": 2,
  },
  "value": 1,
}
`;

exports[`generateLinkedList should generate linked list from different set of values object snapshot 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": "c",
    },
    "value": "b",
  },
  "value": "a",
}
`;

exports[`generateLinkedList should generate linked list from set of values object snapshot 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": 3,
    },
    "value": 2,
  },
  "value": 1,
}
`;
